### PR TITLE
test: 87% coverage of Per diem page

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 336c4f0</samp>

This pull request adds more unit tests for the `add-edit-per-diem` page component in the Fyle mobile app. It creates a new file `add-edit-per-diem-5.page.spec.ts` that contains a function `TestCases5` with various test scenarios. It also modifies the existing file `add-edit-per-diem.page.setup.spec.ts` to import and invoke the `TestCases5` function.

### <samp>🤖 Generated by Copilot at 336c4f0</samp>

> _`TestCases5` adds_
> _more unit tests for the page_
> _autumn of bugs ends_

### <samp>🤖 Generated by Copilot at 336c4f0</samp>

*  Add a new file `add-edit-per-diem-5.page.spec.ts` that contains unit tests for the `add-edit-per-diem` page component (F0)
*  Import and invoke the `TestCases5` function from the new file in the existing setup file `add-edit-per-diem.page.setup.spec.ts` to execute the test cases as part of the test suite ([link](https://github.com/fylein/fyle-mobile-app/pull/2326/files?diff=unified&w=0#diff-113c190c8d5e8c3f8c98cf1e699be2562159cd2ba8d04e2e19a31ef5b74d35c2R41),[link](https://github.com/fylein/fyle-mobile-app/pull/2326/files?diff=unified&w=0#diff-113c190c8d5e8c3f8c98cf1e699be2562159cd2ba8d04e2e19a31ef5b74d35c2R289))

## Clickup
https://app.clickup.com/t/85ztk93f1

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        custom_inputs: new FormArray([]),
+        duplicate_detection_reason: [],
+        billable: [],
+        costCenter: [],
+        project_dependent_fields: formBuilder.array([]),
+        cost_center_dependent_fields: formBuilder.array([]),
+      });
+    }));
+
+    it('reloadCurrentRoute(): should reload the current load', fakeAsync(() => {
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts:142]: should reload the current route?

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+          ...perDiemFormValuesData10,
+        });
+        spyOn(component, 'goBack');
+        spyOn(component, 'reloadCurrentRoute');
+      });
+
+      it('should add expense and go back if form and payment mode is valid', () => {
+        spyOn(component, 'checkIfInvalidPaymentMode').and.returnValue(of(false));
+        component.saveAndNewExpense();
+        expect(component.addExpense).toHaveBeenCalledOnceWith('SAVE_AND_NEW_PER_DIEM');
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts:164]: Should we move these `redirectedFrom` values to enums?

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        expect(component.editExpense).not.toHaveBeenCalled();
+        expect(component.goBack).not.toHaveBeenCalled();
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts:166]: The test description says that `it should go back`, but we are testing `goBack.not.toHaveBeenCalled`

Am I missing something here?

**Review by OmkarJ13 — CHANGES_REQUESTED** [review]

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts (hunk 3 above)
Comment (suyashpatil78) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem-5.page.spec.ts:166]: Instead of it should go back it should be reload current route, corrected it

**Review by Dimple16 — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]

**Pull request merged**